### PR TITLE
2025 Wfs Skeleton and Updates to Data Wfs (2024,2023,2022)

### PR DESCRIPTION
**Pull request by AdrianoDee**
#### PR description:

This PR proposes the addition of 2025 wfs to the matrix. For the moment they are a copy of what we run currently for 2024 with the idea that once 2025 parameters (conditions, HLT) come  we will simply need to fill the dummy ones (keeping them in sync one with the other). Here I add also:

- `@relval2025` pointing to `GRun`;
- `auto:phase1_2025_*` symbolic GTs;

The era is `Era_2025` that for now is simply a copy of `Era_2024`. A TTbar 2025 noPU wf is added to the limited matrix to be tested with PRs.

The rationale is that I would like to avoid (with some anticipation) that we fall again in a situation such as https://github.com/cms-sw/cmssw/issues/44028 or https://github.com/cms-sw/cmssw/issues/41410. 

Additional changes:
- 2023 data wfs GT moved to the offline one (`auto:run3_data_relval`);
- 2024 data wfs added to the PR tests reducing the number of 2022,2023 wfs;
- small fixes for `ZeroBias` 2024 wfs to have the RECO datatier produced for them (as it was in 2023 and 2022) and to run the proper DQM sequence.
- 2024 HLT moved to `relval2025` to keep `GRun` tested;
- `relval_2024` moved to `Fake2`;
- 2023 data wfs HLT moved to `relval_2023` (`Fake2`);

#### PR validation:

New wfs run.

### CODE REVIEW

**Comment by AdrianoDee** [comment]
please test

File: Configuration/HLT/python/autoHLT.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
   'relval2022' : 'Fake2',
   'relval2023' : 'Fake2',
   'relval2024' : 'GRun',
```
Comment (mmusich) [inline Configuration/HLT/python/autoHLT.py:15]: are we going to get 2025 relvals to run in IBs and PRs?

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py:15]: IBs no, PRs I've added a single ttbar noPu.

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py:15]: Meaning: I didn't add the in this PR but it would be doable.

Comment (mmusich) [reply Configuration/HLT/python/autoHLT.py:15]: OK, I let you decide how to move forward. I assume at some point it will be necessary in any case and perhaps we could start doing it now.

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py:15]: I've thought about this and it depends a bit also on how @cms-sw/alca-l2 foresee the evolution of 2024 GTs given this impact the release validations. Currently we run only 2024 wfs and, as is, this allows us to validate both the "latest" 2024 HLT (now `GRun`) and the latest symbolic GT. If we move the 2024 to `Fake` and the 2025 is promoted to `GRun` then I would still like to validate the latter moving the RelVals production to 2025 MC. But then the risk would be that we wouldn't validate the latest 2024 GTs (unless `auto:phase1_2025_*` are explicitly kept in sync or forced to be so in `autoCond.py`). So if AlCa tells me that for 2024 we are more or less done, ok then let's move everything to (2024-like) 2025 and we will move also the RelVal production to it.

Comment (mmusich) [reply Configuration/HLT/python/autoHLT.py:15]: > But then the risk would be that we wouldn't validate the latest 2024 GTs 

mmh, what's the need to validate 2024 GTs in *this* cycle? I suppose they need to be backported anyway in 14.0.X where we don't plan to decommission the 2024 menu.

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py:15]: Just because we run release validations mostly on this cycle and not on `14_0_X` (or `14_1_X`). I see it as a way of profiting of the regular scrutiny of validators to check also the new GTs. Similarly to what happened for HLT being indirectly checked on `14_1_X` during this whole period thanks to `@relval2024`.

Comment (mmusich) [reply Configuration/HLT/python/autoHLT.py:15]: we're not going to re-run the 2024 HLT in any other release than 14.0.X in production. I don't see the reason to validate the 2024 HLT in this cycle. Of course "offline" validators can continue to validate offline reconstruction.

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py:15]: This is clear, I wanted to understand if it's anyway useful to keep an eye on the 2024 GT (for the "offline" validations).

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py:15]: If it's not, I'll move everything to 2025 and that's it.

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py]: I've moved `@relval2024` to `Fake` and added a couple of 2025 wfs to IB tests.

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
please test

**Comment by mmusich** [comment]
@AdrianoDee 

Screenshot from 2024-10-23 09-45-15


```
| 141.044 	| Run2023D JetMET0 	| run3_hlt_relval 	| Run3_2023 	| HLT@relval2024 	|  
| 141.042 	| Run2023D ZeroBias 	| run3_hlt_relval 	| Run3_2023 	| HLT@relval2024 	|  	
| 141.046 	| Run2023D EGamma0 	| run3_hlt_relval 	| Run3_2023 	| HLT@relval2024 	|  	
```

Two things:
   * shall we move 2023 data relvals out of the short matrix and put the corresponding 2024 instead?
   * should these 2024 data relvals use `@relval2025` to keep running a non-fake HLT menu?

**Comment by AdrianoDee** [comment]
Yes, I was thinking to do it in a separate PR. But I could use this chance indeed.

File: Configuration/HLT/python/autoHLT.py (hunk 1 above)
Comment (perrotta) [reply Configuration/HLT/python/autoHLT.py]: > This is clear, I wanted to understand if it's anyway useful to keep an eye on the 2024 GT (for the "offline" validations).

2024 pp GTs are supposed to be "completed" by now.
There is still some work in progress for the 2024 HI GTs, but they are validated via different workflows than the ones addressed here.

**Comment by perrotta** [comment]
@AdrianoDee I assume that as soon as you sign this PR for pdmv also other groups can consider this as concluded from your part and can also sign it

**Comment by AdrianoDee** [comment]
+pdmv

@perrotta I think we were waiting each-other (:

**Comment by perrotta** [comment]
+alca

**Comment by mmusich** [comment]
> Yes, I was thinking to do it in a separate PR. But I could use this chance indeed.

so this won't happen here?

**Comment by mmusich** [comment]
hold

File: Configuration/AlCa/python/autoCond.py [hunk 2]
```diff
@@ -105,6 +105,12 @@
     'phase2_realistic'             :    '141X_mcRun4_realistic_v3'
 }
 
+### 2025 conditions copy of 2024 conditions on 17/10/2024
+### to be updated when 2025 conditions come
```
[the remark this answers is not in the document]
Comment (AdrianoDee) [reply Configuration/AlCa/python/autoCond.py]: Took the chance to implement this

**Comment by AdrianoDee** [comment]
> so this won't happen here?

Sorry, it had slipped out of my mind. I've moved the 2024 data wfs to `relval2025` and added them to the `limited` matrix, keeping anyway a pair of 2023/2022 wfs.

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
please test

**Comment by mmusich** [comment]
unhold

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 3]
```diff
@@ -2772,8 +2773,8 @@ def lhegensim2018ml(fragment,howMuch):
 steps['RECODR3_2024']=merge([{'--era':'Run3_2024'},steps['RECODR3']])
 
 steps['RECODR3_reHLT_2022']=merge([{'--conditions':'auto:run3_data_relval', '--hltProcess':'reHLT'},steps['RECODR3']])
-steps['RECODR3_reHLT_2023']=merge([{'--conditions':'auto:run3_data_prompt_relval', '--hltProcess':'reHLT'},steps['RECODR3_2023']])
-steps['RECODR3_reHLT_2023B']=merge([{'--conditions':'auto:run3_data_prompt_relval', '--hltProcess':'reHLT'},steps['RECODR3']])
+steps['RECODR3_reHLT_2023']=merge([{'--conditions':'auto:run3_data_relval', '--hltProcess':'reHLT'},steps['RECODR3_2023']])
```
Comment (AdrianoDee) [inline Configuration/PyReleaseValidation/python/relval_steps.py:2776]: @perrotta I took the chance to move the 2023 to the offline GT as we were discussing. The online one will be tested by 2024 wfs.

**Comment by AdrianoDee** [comment]
I have taken the chance to add some other fixes and updates to the matrix. See the description.

**Comment by AdrianoDee** [comment]
please test

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 4]
```diff
@@ -2232,13 +2232,14 @@ def lhegensim2018ml(fragment,howMuch):
...
 
-hltKey2024='relval2024' # currently points to GRUN (hacky solution to keep running GRun on real data)
-steps['HLTDR3_2023']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2024,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2023'},steps['HLTD'] ] )
+steps['HLTDR3_2023']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2023,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2023'},steps['HLTD'] ] )
 
-steps['HLTDR3_2023B']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2024,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3'},steps['HLTD'] ] )
+steps['HLTDR3_2023B']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2023,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3'},steps['HLTD'] ] )
 
-steps['HLTDR3_2024']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2024,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2024'},steps['HLTD'] ] )
+steps['HLTDR3_2024']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2023,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2024'},steps['HLTD'] ] )
```
Comment (Martin-Grunewald) [inline Configuration/PyReleaseValidation/python/relval_steps.py:2242]: This could now be ```hltKey2024``` as that is now ```Fake2``` and the line is about ```HLTDR3_2024``` (?)

Comment (AdrianoDee) [reply Configuration/PyReleaseValidation/python/relval_steps.py]: I think we can move to `hltKey2025` for the moment just to run the `GRun` (taking the suggestion in https://github.com/cms-sw/cmssw/pull/46445#issuecomment-2431187861). I've just pushed that change.

**Comment by AdrianoDee** [comment]
please test

File: Configuration/HLT/python/autoHLT.py [hunk 5]
```diff
@@ -12,7 +12,8 @@
   'relval2018' : 'Fake2',
   'relval2022' : 'Fake2',
   'relval2023' : 'Fake2',
-  'relval2024' : 'GRun',
+  'relval2024' : 'Fake2',
+  'relval2025' : 'GRun',
   'relval2026' : '75e33',
```
Comment (Martin-Grunewald) [inline Configuration/HLT/python/autoHLT.py:17]: Hmm, 2026 is rather confusing by now as it is by now still phase1, rather than phase2 which should now be 2030?
When is that update redefining 2026 foreseen?

Comment (AdrianoDee) [reply Configuration/HLT/python/autoHLT.py:17]: Indeed, it is the next item on my todo list.

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 6]
```diff
@@ -2232,13 +2232,14 @@ def lhegensim2018ml(fragment,howMuch):
...
 
-hltKey2024='relval2024' # currently points to GRUN (hacky solution to keep running GRun on real data)
-steps['HLTDR3_2023']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2024,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2023'},steps['HLTD'] ] )
+steps['HLTDR3_2023']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2023,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2023'},steps['HLTD'] ] )
 
-steps['HLTDR3_2023B']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2024,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3'},steps['HLTD'] ] )
+steps['HLTDR3_2023B']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2023,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3'},steps['HLTD'] ] )
 
-steps['HLTDR3_2024']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2024,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2024'},steps['HLTD'] ] )
+steps['HLTDR3_2024']=merge( [ {'-s':'L1REPACK:Full,HLT:@%s'%hltKey2025,},{'--conditions':'auto:run3_hlt_relval'},{'--era' : 'Run3_2024'},steps['HLTD'] ] )
```
[the remark this answers is not in the document]
Comment (AdrianoDee) [reply Configuration/PyReleaseValidation/python/relval_steps.py:2242]: Being this data wfs I think we could roll it back only when 2025 data wfs come. To have 2025 before that (to have 2024 wfs running with `@relval2024`, the only possibility I see is to have some "fake" 2025 data workflows running actually on 2024 data. But I think this could be even more confusing. Note that for MC instead the key used for 2024 workflows is `@relval2024` and 2025 wfs will exercise the GRun.

**Comment by AdrianoDee** [comment]
(files are being copied to T2_CH_CERN)

**Comment by AdrianoDee** [comment]
please test

**Comment by AdrianoDee** [comment]
please test with https://github.com/cms-sw/cmssw/pull/46559

**Comment by AdrianoDee** [comment]
+pdmv

File: Configuration/HLT/python/autoHLT.py (hunk 5 above)
Comment (mmusich) [reply Configuration/HLT/python/autoHLT.py:17]: See also https://github.com/cms-sw/cmssw/issues/46028

**Comment by mmusich** [comment]
+hlt
   * for https://github.com/AdrianoDee/cmssw/commit/c9e7be9a5d0aafc0b0ce30e767471986140bc251
   * PR according to description and follow-up review

File: Configuration/PyReleaseValidation/python/relval_data_highstats.py [hunk 7]
```diff
@@ -12,50 +12,56 @@
...
     for p_n,pd in enumerate(pds_2024):
         for e_key,evs in event_steps_dict.items():
             if "10k" == e_key: # already defined in relval_standard
                 continue   
-            wf_number = base_wf_number_2024
+            wf_number = base_wf
             wf_number = wf_number + offset_era * e_n
             wf_number = wf_number + offset_pd * p_n
-            wf_number = wf_number + offset_events * evs 
+            wf_number = wf_number + 0.0001 * 0.01
```
Comment (perrotta) [inline Configuration/PyReleaseValidation/python/relval_data_highstats.py:24]: Just curious: why did you move to the hardcoded numbers rather than sticking on the parameterized ones? Why do you do so for all eras but the `eras_2022_2` one?

Comment (AdrianoDee) [reply Configuration/PyReleaseValidation/python/relval_data_highstats.py:24]: It was a stupid copy paste. Thanks for spotting it. Just reverted.

**Comment by AdrianoDee** [comment]
please test with https://github.com/cms-sw/cmssw/pull/46559

**Comment by AdrianoDee** [comment]
please abort

**Comment by AdrianoDee** [comment]
please test with https://github.com/cms-sw/cmssw/pull/46559

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by perrotta** [comment]
+alca

**Comment by Moanwar** [comment]
+Upgrade

**Comment by mandrenguyen** [comment]
+1
